From 8759995c0ccea0d6fac94b0f7e9dd85af27f6e87 Mon Sep 17 00:00:00 2001
From: Petr Lautrbach <plautrba@redhat.com>
Date: Thu, 7 Jan 2016 15:57:42 +0100
Subject: [PATCH] restorecond: treat root as a regular user

When restorecond is run with -u, root should be treated as a regular user

Resolves: rhbz#1281877
---
 restorecond/restorecond.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/restorecond/restorecond.c b/restorecond/restorecond.c
index 89f5d97..c71a574 100644
--- a/restorecond/restorecond.c
+++ b/restorecond/restorecond.c
@@ -205,9 +205,11 @@ int main(int argc, char **argv)
 		exitApp("getpwuid");
 
 	homedir = pwd->pw_dir;
+
+	if (run_as_user)
+		return server(master_fd, user_watch_file);
+
 	if (uid != 0) {
-		if (run_as_user)
-			return server(master_fd, user_watch_file);
 		if (start() != 0)
 			return server(master_fd, user_watch_file);
 		return 0;
-- 
1.8.3.1

